src/core/a-entity.js

```diff
@@ -301,14 +302,52 @@ var proto = Object.create(ANode.prototype, {
     }
   },
 
+  getMixedInComponents: {
+    value: function () {
+      var mixinEls = this.mixinEls;
+      var i;
+      var components = [];
+      for (i = 0; i < mixinEls.length; ++i) {
+        var keys = Object.keys(mixinEls[i].attrs);
+        keys.forEach(function (key) {
+          components.push(key);
+        });
+      }
+      return components;
+    }
+  },
+
   updateComponents: {
     value: function () {
-      var self = this;
-      var allComponents = Object.keys(components);
-      allComponents.forEach(updateComponent);
-      function updateComponent (name) {
-        var elValue = self.getAttribute(name);
-        self.updateComponent(name, elValue);
+      var attrs = {};
+      var componentName;
+      var i;
+      if (!this.hasLoaded) { return; }
+      var elComponents = Object.keys(this.components);
```

Comment:
Some `forEach` and less intermediate vars might make this function easier to read as well.